src/schema/v2/previewSavedSearch.ts

```diff
@@ -99,13 +101,72 @@ const PreviewSavedSearchType = new GraphQLObjectType<any, ResolverContext>({
       type: new GraphQLNonNull(
         new GraphQLList(new GraphQLNonNull(SearchCriteriaLabel))
       ),
-      resolve: (_) => {
-        return mockSuggestedFilters
+      resolve: async (_root, _args, { filterArtworksLoader }) => {
+        const suggestedFiltersByArtist: SearchCriteriaLabel[][] = await Promise.all(
+          _root.artistIDs.map(
+            async (artistSlug) =>
+              await getSuggestedFiltersByArtistSlug(
+                artistSlug,
+                filterArtworksLoader
+              )
```

Comment:
No need to await the result if it's returned immediately.

```suggestion
           (artistSlug) =>
              getSuggestedFiltersByArtistSlug(
                artistSlug,
                filterArtworksLoader
              )
```

Reply:
good point, i am doing it within the function already